service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/OrdersNeededAboutToSubmitCallbackController.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import io.swagger.annotations.Api;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Api
+@RestController
+@RequestMapping("/callback/orders-needed")
+public class OrdersNeededAboutToSubmitCallbackController {
+
+    @PostMapping("/about-to-submit")
+    @SuppressWarnings("unchecked")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(
+        @RequestBody CallbackRequest callbackrequest) {
+        String epo = "EMERGENCY_PROTECTION_ORDER";
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        Optional<List<String>> orderType = Optional.ofNullable((Map<String, Object>) data.get("orders"))
```

Comment:
Why not to do:

```
Optional.ofNullable((Map<String, Object>) data.get("orders"))
  .map(orders -> (List<String>) orders.get("orderType"))
  .ifPresent(orderTypes -> {
    ... // orderTypes here should be of type List<String>
  })
```

Doing `.toString()` is really nasty. It relies on an implementation of `toString` in `Optional` and `List` classes that can change. Not likely but can change.